wolfcrypt/src/port/nxp/se050_port.c

```diff
@@ -1268,4 +1269,168 @@ int se050_ed25519_verify_msg(const byte* signature, word32 signatureLen,
+    sss_object_t            ref_public_key;
+    sss_object_t            deriveKey;
+    sss_derive_key_t        ctx_derive_key;
+    int                     keyId;
+    int                     keySize = CURVE25519_KEYSIZE;
+    size_t                  ecdhKeyLen = keySize;
+    size_t                  ecdhKeyBitLen = keySize;
+    int                     ret = WC_HW_E;
+
+    if (cfg_se050_i2c_pi == NULL) {
+        return WC_HW_E;
+    }
+
+    if (private_key->keyId <= 0 || public_key->keyId <= 0) {
+        return BAD_FUNC_ARG;
+    }
+
+    if (wolfSSL_CryptHwMutexLock() != 0) {
+        return BAD_MUTEX_E;
+    }
+
+    status = sss_key_store_context_init(&host_keystore, cfg_se050_i2c_pi);
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_allocate(&host_keystore, 60);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_init(&ref_public_key, &host_keystore);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_get_handle(&ref_public_key, public_key->keyId);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_context_init(&host_keystore_2, cfg_se050_i2c_pi);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_allocate(&host_keystore_2, 60);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_init(&ref_private_key, &host_keystore_2);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_get_handle(&ref_private_key, private_key->keyId);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_init(&deriveKey, hostKeyStore);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        keyId = se050_allocate_key(SE050_KEYID_ANY);
+
+        status = sss_key_object_allocate_handle(&deriveKey,
+            keyId,
+            kSSS_KeyPart_Default,
+            kSSS_CipherType_AES,
+            ecdhKeyLen,
+            kKeyObject_Mode_Transient);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_derive_key_context_init(&ctx_derive_key, cfg_se050_i2c_pi,
+                                    &ref_private_key, kAlgorithm_SSS_ECDH,
+                                    kMode_SSS_ComputeSharedSecret);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_derive_key_dh(&ctx_derive_key, &ref_public_key, &deriveKey);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        size_t outlenSz = (size_t)*outlen;
+        status = sss_key_store_get_key(hostKeyStore, &deriveKey, out, &outlenSz,
+                                                                &ecdhKeyBitLen);
+        *outlen = outlenSz;
+    }
+    if (ctx_derive_key.session != NULL)
+        sss_derive_key_context_free(&ctx_derive_key);
+    if (deriveKey.keyStore != NULL)
+        sss_key_object_free(&deriveKey);
+
+    if (status == kStatus_SSS_Success)
+        ret = 0;
+    else
+        ret = WC_HW_E;
```

Comment:
Error case cleanup is not handled correctly. Can you please use `se050_ecc_shared_secret` as a template?